nemo/collections/nlp/modules/common/megatron/transformer.py

```diff
@@ -1435,6 +1436,8 @@ def forward(
         if get_key_value:
             output = [output, presents]
 
+        if self.is_adapter_available():
```

Comment:
In order to implement this, you'll probably need to override forward_single_adapter method and call the correct adapter instead of chaining two adapters - which can be done more easily with module adapters. Look into some of the methods of this class and see what to override to get your designed logic.

Reply:
Updated to include two adapters and using `forward_single_adapter()`